gladier/tools/globus/compute.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import hashlib
+import json
+import typing as t
+from pathlib import Path
+
+from globus_compute_sdk import Client as ComputeClient
+from globus_compute_sdk.serialize import ComputeSerializer
+
+from gladier import GladierActionState, JSONObject
+
+_cache_file = Path.home() / ".gladier_function_id_cache.json"
+
+
+def hash_for_callable(cllable: t.Callable[[t.Any], t.Any]) -> str:
+    serializer = ComputeSerializer()
+    serial = serializer.serialize(cllable)
+    sha256 = hashlib.sha256()
+    sha256.update(bytes(serial, "utf-8"))
+    return sha256.hexdigest()
+
+
+def get_globus_compute_function_cache() -> t.Dict[str, t.Tuple[str, str]]:
+    try:
+        with open(_cache_file, "r") as f:
+            cache = json.load(f)
+            return cache
+    except FileNotFoundError:
+        return {}
+
+
+def save_globus_compute_function_cache(cache: t.Dict[str, t.Tuple[str, str]]) -> None:
+    with open(_cache_file, "w") as f:
+        json.dump(cache, f, indent=2)
+
+
+def register_function_with_compute(cllable: t.Callable[[t.Any], t.Any]) -> str:
+    compute_client = ComputeClient()
+    cllable_id = compute_client.register_function(cllable)
+    return cllable_id
```

Comment:
This needs to be compliant with Gladier's current storage system for registering functions.